grpc-gateway/service/createAndDeleteResource_test.go

```diff
@@ -281,15 +270,14 @@ func TestCreateAndDeleteResource(t *testing.T) {
 	subClient, subID := subscribeToAllEvents(t, ctx, c, correlationID)
 	const switchID = "1"
 
-	isDiscoveryResourceBatchObservable := test.IsDiscoveryResourceBatchObservable(ctx, t, deviceID)
-
 	for i := 0; i < 5; i++ {
 		fmt.Printf("iteration %v\n", i)
+		time.Sleep(time.Millisecond * 500)
```

Comment:
Sleep after print seems weird, is the placement correct? Usually we delay after a create/delete call.

Reply:
As it is async, unpublish events can still be processed during resource creation. (some time the iotivity-lite test fails so I needed add there some time to get idle state)